common.yml

```diff
@@ -20,7 +20,7 @@ variables:
   policy: pull
 
 .ruby:
-  image: ruby:$RUBY_VERSION
+  image: ${RUBY_PLATFORM}:$RUBY_VERSION
```

Comment:
Prefer to use `rules` to change it to constraint the value in `ruby` or `jruby`

```yaml
job:
  # ...
  rules:
    rules:
    - if: $RUBY_PLATFORM == "jruby"
      variables:
          RUBY_IMAGE: jruby
```

Maybe migrate to GitLab Components will better for security
https://docs.gitlab.com/ee/ci/components/

Reply:
https://docs.gitlab.com/ee/ci/yaml/index.html#specinputs

The `spec:inputs` seems to work correctly for non-components. 

> GitLab Component only work for same GitLab server or use mirror (Premium or higher)